README.md
# @36node/bus-op-sdk

[![version][0]][1] [![downloads][2]][3]

主要负责报警处置和工单系统。

## Install

```bash
yarn add @36node/bus-op-sdk
```

## Usage

```js
const SDK = require("@36node/bus-op-sdk");

const sdk = new SDK({ base: "http://localhost:3000" });
await sdk.alert.listAlerts();
await sdk.fault.listFaults();
await sdk.warning.listWarnings();
```

## Test

```sh
yarn test:int
```

## Contributing

1. Fork it!
2. Create your feature branch: `git checkout -b my-new-feature`
3. Commit your changes: `git commit -am 'Add some feature'`
4. Push to the branch: `git push origin my-new-feature`
5. Submit a pull request :D

## Author

**module** © [36node](https://github.com/36node), Released under the [MIT](./LICENSE) License.

Authored and maintained by 36node with help from contributors ([list](https://github.com/36node/module/contributors)).

> GitHub [@36node](https://github.com/36node)

[0]: https://img.shields.io/npm/v/@36node/bus-op-sdk.svg?style=flat
[1]: https://npmjs.com/package/@36node/bus-op-sdk
[2]: https://img.shields.io/npm/dm/@36node/bus-op-sdk.svg?style=flat
[3]: https://npmjs.com/package/@36node/bus-op-sdk


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.4.14](https://github.com/36node/bus-op-sdk-js/compare/v0.4.13...v0.4.14) (2020-04-28)

### [0.4.13](https://github.com/36node/bus-op-sdk-js/compare/v0.4.12...v0.4.13) (2020-03-27)

### [0.4.12](https://github.com/36node/bus-op-sdk-js/compare/v0.4.11...v0.4.12) (2020-03-25)

### [0.4.11](https://github.com/36node/bus-op-sdk-js/compare/v0.4.10...v0.4.11) (2020-03-14)

### [0.4.10](https://github.com/36node/bus-op-sdk-js/compare/v0.4.9...v0.4.10) (2020-03-14)


### Features

* add template ([5e252d2](https://github.com/36node/bus-op-sdk-js/commit/5e252d2))

### [0.4.9](https://github.com/36node/bus-op-sdk-js/compare/v0.4.8...v0.4.9) (2020-01-31)


### Bug Fixes

* update op-sdk swagen ([e2154a9](https://github.com/36node/bus-op-sdk-js/commit/e2154a9))

### [0.4.8](https://github.com/36node/bus-op-sdk-js/compare/v0.4.7...v0.4.8) (2019-10-28)


### Features

* update ([b5d2be6](https://github.com/36node/bus-op-sdk-js/commit/b5d2be6))

### [0.4.7](https://github.com/36node/bus-op-sdk-js/compare/v0.4.6...v0.4.7) (2019-10-14)


### Bug Fixes

* expo mjs extension err ([45c6ac7](https://github.com/36node/bus-op-sdk-js/commit/45c6ac7))

### [0.4.6](https://github.com/36node/bus-op-sdk-js/compare/v0.4.5...v0.4.6) (2019-08-23)



### [0.4.5](https://github.com/36node/bus-op-sdk-js/compare/v0.4.4...v0.4.5) (2019-08-22)


### Features

* add description ([8e2da80](https://github.com/36node/bus-op-sdk-js/commit/8e2da80))

### [0.4.4](https://github.com/36node/bus-op-sdk-js/compare/v0.4.3...v0.4.4) (2019-08-22)



### [0.4.3](https://github.com/36node/bus-op-sdk-js/compare/v0.4.2...v0.4.3) (2019-08-07)


### Features

* add query to alert summary ([351e2b6](https://github.com/36node/bus-op-sdk-js/commit/351e2b6))

### [0.4.2](https://github.com/36node/bus-op-sdk-js/compare/v0.4.1...v0.4.2) (2019-07-29)



### [0.4.1](https://github.com/36node/bus-op-sdk-js/compare/v0.4.0...v0.4.1) (2019-07-24)


### Bug Fixes

* int test ([768b325](https://github.com/36node/bus-op-sdk-js/commit/768b325))



## [0.4.0](https://github.com/36node/bus-op-sdk-js/compare/v0.3.0...v0.4.0) (2019-07-24)


### Features

* tickets mock ([9a62379](https://github.com/36node/bus-op-sdk-js/commit/9a62379))



## [0.3.0](https://github.com/36node/bus-op-sdk-js/compare/v0.2.0...v0.3.0) (2019-07-21)


### Features

* ticket mock ([1faeefe](https://github.com/36node/bus-op-sdk-js/commit/1faeefe))



## [0.2.0](https://github.com/36node/bus-op-sdk-js/compare/v0.1.3...v0.2.0) (2019-07-17)


### Bug Fixes

* remove count ([8f75f1d](https://github.com/36node/bus-op-sdk-js/commit/8f75f1d))


### Features

* ticket mockup ([57d25b4](https://github.com/36node/bus-op-sdk-js/commit/57d25b4))



### [0.1.3](https://github.com/36node/bus-op-sdk-js/compare/v0.1.2...v0.1.3) (2019-07-02)



### [0.1.2](https://github.com/36node/bus-op-sdk-js/compare/v0.1.1...v0.1.2) (2019-06-30)



### [0.1.1](https://github.com/36node/bus-op-sdk-js/compare/v0.1.0...v0.1.1) (2019-06-26)



## [0.1.0](https://github.com/36node/bus-op-sdk-js/compare/v0.0.5...v0.1.0) (2019-06-24)


### Features

* add warnings type mock ([#4](https://github.com/36node/bus-op-sdk-js/issues/4)) ([9a0db76](https://github.com/36node/bus-op-sdk-js/commit/9a0db76))



### [0.0.5](https://github.com/36node/bus-op-sdk-js/compare/v0.0.4...v0.0.5) (2019-06-14)


### Bug Fixes

* typings in npm package ([1c10684](https://github.com/36node/bus-op-sdk-js/commit/1c10684))



### [0.0.4](https://github.com/36node/bus-op-sdk-js/compare/v0.0.3...v0.0.4) (2019-06-14)



### [0.0.3](https://github.com/36node/bus-op-sdk-js/compare/v0.0.2...v0.0.3) (2019-06-13)



### [0.0.2](https://github.com/36node/bus-op-sdk-js/compare/v0.0.1...v0.0.2) (2019-06-13)



### 0.0.1 (2019-06-13)


mock/company.js
module.exports = [
  {
    convoys: [
      {
        lines: [
          { stations: [], name: "947" },
          { stations: [], name: "67" },
          { stations: [], name: "94" },
        ],
        name: "九车队",
      },
      {
        lines: [
          { stations: [], name: "931" },
          { stations: [], name: "911" },
          { stations: [], name: "927" },
          { stations: [], name: "824" },
          { stations: [], name: "138" },
          { stations: [], name: "浦西滨江1路" },
          { stations: [], name: "一车队备车D" },
        ],
        name: "一车队",
      },
      {
        lines: [
          { stations: [], name: "17" },
          { stations: [], name: "96" },
          { stations: [], name: "736" },
          { stations: [], name: "1212" },
          { stations: [], name: "1216" },
        ],
        name: "三车队",
      },
      {
        lines: [
          { stations: [], name: "973" },
          { stations: [], name: "41" },
          { stations: [], name: "隧道夜宵一线" },
          { stations: [], name: "734" },
          { stations: [], name: "隧道二线" },
          { stations: [], name: "大桥一线" },
        ],
        name: "二车队",
      },
      { lines: [{ stations: [], name: "申川专线" }], name: "六车队" },
      { lines: [{ stations: [], name: "929" }], name: "七车队" },
      {
        lines: [{ stations: [], name: "926" }, { stations: [], name: "64" }],
        name: "八车队",
      },
      {
        lines: [
          { stations: [], name: "937" },
          { stations: [], name: "951" },
          { stations: [], name: "510" },
        ],
        name: "十车队",
      },
    ],
    name: "巴士四公司",
    updatedAt: "2018-11-22T03:41:10.075Z",
    createdAt: "2018-11-22T03:38:18.861Z",
    id: "5bf624aa0ebb6f55ab5c56a0",
  },
  {
    convoys: [
      {
        lines: [
          { stations: [], name: "新泾1路" },
          { stations: [], name: "九车队备车C" },
          { stations: [], name: "158" },
          { stations: [], name: "72" },
          { stations: [], name: "320" },
          { stations: [], name: "54" },
          { stations: [], name: "88" },
          { stations: [], name: "316" },
          { stations: [], name: "121" },
          { stations: [], name: "1217" },
          { stations: [], name: "855" },
        ],
        name: "九车队",
      },
      {
        lines: [
          { stations: [], name: "825" },
          { stations: [], name: "八车队备车C" },
          { stations: [], name: "91" },
          { stations: [], name: "闵行28路" },
          { stations: [], name: "176" },
          { stations: [], name: "141" },
          { stations: [], name: "173" },
          { stations: [], name: "197" },
        ],
        name: "八车队",
      },
      {
        lines: [
          { stations: [], name: "757" },
          { stations: [], name: "十二车队备车C" },
          { stations: [], name: "754" },
          { stations: [], name: "739" },
          { stations: [], name: "765" },
        ],
        name: "十二车队",
      },
      {
        lines: [
          { stations: [], name: "十一车队备车C" },
          { stations: [], name: "519" },
          { stations: [], name: "748" },
          { stations: [], name: "57" },
          { stations: [], name: "1207" },
          { stations: [], name: "48" },
          { stations: [], name: "709" },
        ],
        name: "十一车队",
      },
      {
        lines: [
          { stations: [], name: "十车队备车C" },
          { stations: [], name: "721" },
          { stations: [], name: "闵行36路" },
          { stations: [], name: "76" },
        ],
        name: "十车队",
      },
      { lines: [{ stations: [], name: "91" }], name: "十五车队" },
      {
        lines: [
          { stations: [], name: "561" },
          { stations: [], name: "224" },
          { stations: [], name: "252" },
        ],
        name: "三车队",
      },
      {
        lines: [
          { stations: [], name: "923" },
          { stations: [], name: "846" },
          { stations: [], name: "62" },
          { stations: [], name: "738" },
          { stations: [], name: "737" },
        ],
        name: "六车队",
      },
      {
        lines: [
          { stations: [], name: "717" },
          { stations: [], name: "63" },
          { stations: [], name: "105" },
          { stations: [], name: "136" },
          { stations: [], name: "948" },
          { stations: [], name: "768" },
        ],
        name: "二车队",
      },
      {
        lines: [{ stations: [], name: "909" }, { stations: [], name: "950" }],
        name: "七车队",
      },
      {
        lines: [
          { stations: [], name: "859" },
          { stations: [], name: "750" },
          { stations: [], name: "117" },
          { stations: [], name: "193" },
          { stations: [], name: "1227" },
          { stations: [], name: "1231" },
          { stations: [], name: "837" },
          { stations: [], name: "838" },
          { stations: [], name: "856" },
        ],
        name: "四车队",
      },
      {
        lines: [
          { stations: [], name: "724" },
          { stations: [], name: "743" },
          { stations: [], name: "216" },
          { stations: [], name: "106" },
          { stations: [], name: "112" },
        ],
        name: "五车队",
      },
      {
        lines: [
          { stations: [], name: "71" },
          { stations: [], name: "71区间" },
          { stations: [], name: "1250路（71路支线1）" },
          { stations: [], name: "1251路（71路支线2）" },
          { stations: [], name: "835" },
        ],
        name: "一车队",
      },
      { lines: [{ stations: [], name: "1250" }], name: "中运量车队" },
      { lines: [{ stations: [], name: "十三车队备车C" }], name: "十三车队" },
    ],
    name: "巴士三公司",
    updatedAt: "2018-11-22T03:41:08.872Z",
    createdAt: "2018-11-22T03:38:15.839Z",
    id: "5bf624a70ebb6f55ab5c566e",
  },
  {
    convoys: [
      { lines: [{ stations: [], name: "50区" }], name: "梅陇" },
      {
        lines: [
          { stations: [], name: "946" },
          { stations: [], name: "87" },
          { stations: [], name: "808" },
        ],
        name: "漕宝车队",
      },
      {
        lines: [
          { stations: [], name: "虹桥枢纽4路" },
          { stations: [], name: "735" },
          { stations: [], name: "732" },
          { stations: [], name: "731" },
          { stations: [], name: "闵行33路" },
        ],
        name: "平阳车队",
      },
      {
        lines: [
          { stations: [], name: "131" },
          { stations: [], name: "122" },
          { stations: [], name: "152" },
          { stations: [], name: "712" },
        ],
        name: "梅陇车队",
      },
      {
        lines: [
          { stations: [], name: "150" },
          { stations: [], name: "162" },
          { stations: [], name: "闵行22路" },
          { stations: [], name: "763" },
          { stations: [], name: "古美环线" },
          { stations: [], name: "1202" },
          { stations: [], name: "747" },
          { stations: [], name: "南站车队备车B" },
        ],
        name: "南站车队",
      },
      {
        lines: [
          { stations: [], name: "92" },
          { stations: [], name: "92B" },
          { stations: [], name: "764" },
          { stations: [], name: "725" },
          { stations: [], name: "703" },
        ],
        name: "龙华车队",
      },
      {
        lines: [{ stations: [], name: "759" }, { stations: [], name: "189" }],
        name: "莘庄车队",
      },
      {
        lines: [
          { stations: [], name: "704" },
          { stations: [], name: "704B" },
          { stations: [], name: "157" },
          { stations: [], name: "718" },
          { stations: [], name: "49" },
        ],
        name: "49路车队",
      },
      {
        lines: [
          { stations: [], name: "56" },
          { stations: [], name: "178" },
          { stations: [], name: "178大站车" },
          { stations: [], name: "820" },
          { stations: [], name: "958" },
          { stations: [], name: "111" },
          { stations: [], name: "770" },
          { stations: [], name: "56区间" },
          { stations: [], name: "707" },
        ],
        name: "龙吴车队",
      },
    ],
    name: "巴士二公司",
    updatedAt: "2018-11-22T03:41:07.083Z",
    createdAt: "2018-11-22T03:38:22.529Z",
    id: "5bf624ae0ebb6f55ab5c56dd",
  },
  {
    convoys: [
      {
        lines: [
          { stations: [], name: "8" },
          { stations: [], name: "28" },
          { stations: [], name: "22" },
          { stations: [], name: "139" },
          { stations: [], name: "868" },
          { stations: [], name: "934" },
        ],
        name: "十二车队",
      },
      {
        lines: [
          { stations: [], name: "13" },
          { stations: [], name: "24" },
          { stations: [], name: "23" },
          { stations: [], name: "939" },
        ],
        name: "十车队",
      },
      {
        lines: [
          { stations: [], name: "15" },
          { stations: [], name: "20" },
          { stations: [], name: "21" },
          { stations: [], name: "19" },
        ],
        name: "八车队",
      },
      {
        lines: [
          { stations: [], name: "6" },
          { stations: [], name: "14" },
          { stations: [], name: "十三车队备车A" },
          { stations: [], name: "746" },
        ],
        name: "十三车队",
      },
      {
        lines: [
          { stations: [], name: "25" },
          { stations: [], name: "37" },
          { stations: [], name: "155" },
        ],
        name: "十四车队",
      },
      {
        lines: [
          { stations: [], name: "大桥三线" },
          { stations: [], name: "966" },
          { stations: [], name: "168" },
          { stations: [], name: "90" },
          { stations: [], name: "528" },
        ],
        name: "六车队",
      },
      {
        lines: [{ stations: [], name: "723" }, { stations: [], name: "842" }],
        name: "四车队",
      },
      {
        lines: [
          { stations: [], name: "870" },
          { stations: [], name: "405" },
          { stations: [], name: "406" },
          { stations: [], name: "124" },
          { stations: [], name: "102" },
        ],
        name: "七车队",
      },
      {
        lines: [
          { stations: [], name: "五车队备车A" },
          { stations: [], name: "115" },
          { stations: [], name: "100" },
          { stations: [], name: "1258" },
          { stations: [], name: "942" },
        ],
        name: "五车队",
      },
      {
        lines: [
          { stations: [], name: "875" },
          { stations: [], name: "70" },
          { stations: [], name: "758" },
          { stations: [], name: "三车队备车A" },
          { stations: [], name: "222" },
          { stations: [], name: "80" },
          { stations: [], name: "79" },
          { stations: [], name: "960" },
          { stations: [], name: "559" },
        ],
        name: "三车队",
      },
      {
        lines: [
          { stations: [], name: "二车队备车A" },
          { stations: [], name: "140" },
          { stations: [], name: "179" },
          { stations: [], name: "66区间" },
          { stations: [], name: "66" },
          { stations: [], name: "97" },
        ],
        name: "二车队",
      },
      {
        lines: [
          { stations: [], name: "99" },
          { stations: [], name: "1218" },
          { stations: [], name: "1228" },
          { stations: [], name: "1226" },
          { stations: [], name: "55" },
          { stations: [], name: "61" },
          { stations: [], name: "1201" },
          { stations: [], name: "1255" },
          { stations: [], name: "1256" },
        ],
        name: "一车队",
      },
      { lines: [{ stations: [], name: "测试分组" }], name: "测试分组" },
    ],
    name: "巴士一公司",
    updatedAt: "2018-11-22T03:40:57.621Z",
    createdAt: "2018-11-22T03:38:18.909Z",
    id: "5bf624aa0ebb6f55ab5c56a1",
  },
  {
    convoys: [
      {
        lines: [
          { stations: [], name: "210" },
          { stations: [], name: "58" },
          { stations: [], name: "185" },
          { stations: [], name: "767" },
          { stations: [], name: "宝山19路" },
          { stations: [], name: "821" },
          { stations: [], name: "宝山85路" },
          { stations: [], name: "151" },
          { stations: [], name: "1209" },
          { stations: [], name: "二车队备车A" },
        ],
        name: "二车队",
      },
      {
        lines: [{ stations: [], name: "187" }, { stations: [], name: "845" }],
        name: "五车队",
      },
      {
        lines: [{ stations: [], name: "705" }, { stations: [], name: "165" }],
        name: "七车队",
      },
      { lines: [{ stations: [], name: "527" }], name: "一车队" },
      {
        lines: [
          { stations: [], name: "78" },
          { stations: [], name: "68" },
          { stations: [], name: "321" },
          { stations: [], name: "70" },
        ],
        name: "三车队",
      },
      {
        lines: [
          { stations: [], name: "727" },
          { stations: [], name: "702" },
          { stations: [], name: "110" },
          { stations: [], name: "528" },
        ],
        name: "六车队",
      },
      {
        lines: [
          { stations: [], name: "728" },
          { stations: [], name: "701" },
          { stations: [], name: "宝山29路" },
          { stations: [], name: "宝山17路" },
          { stations: [], name: "十车队备车E" },
          { stations: [], name: "1610" },
          { stations: [], name: "宝山36路" },
        ],
        name: "十车队",
      },
      {
        lines: [
          { stations: [], name: "812" },
          { stations: [], name: "1601" },
          { stations: [], name: "1612" },
          { stations: [], name: "760" },
          { stations: [], name: "1608" },
          { stations: [], name: "118" },
        ],
        name: "十二车队",
      },
      {
        lines: [
          { stations: [], name: "宝山25路" },
          { stations: [], name: "宝山5路" },
          { stations: [], name: "宝山18路" },
          { stations: [], name: "883" },
          { stations: [], name: "宝山91路" },
          { stations: [], name: "宝山89路" },
          { stations: [], name: "宝山86路" },
          { stations: [], name: "952" },
          { stations: [], name: "811" },
          { stations: [], name: "宝山10路" },
        ],
        name: "十三车队",
      },
      {
        lines: [{ stations: [], name: "711" }, { stations: [], name: "1609" }],
        name: "四车队",
      },
      {
        lines: [
          { stations: [], name: "宝山15路" },
          { stations: [], name: "宝山2路" },
          { stations: [], name: "十一车队备车E" },
          { stations: [], name: "宝山12路" },
          { stations: [], name: "宝山82路" },
          { stations: [], name: "宝山92路" },
          { stations: [], name: "1604" },
          { stations: [], name: "1605" },
          { stations: [], name: "宝山16路" },
          { stations: [], name: "宝山30路" },
          { stations: [], name: "1602" },
          { stations: [], name: "宝山13路" },
        ],
        name: "十一车队",
      },
      { lines: [{ stations: [], name: "宝山11路" }], name: "九车队" },
    ],
    name: "巴士五公司",
    updatedAt: "2018-11-22T03:40:53.851Z",
    createdAt: "2018-11-22T03:38:46.693Z",
    id: "5bf624c60ebb6f55ab5c5895",
  },
];
